pydantic/dataclasses.py

```diff
@@ -190,6 +192,12 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
         Returns:
             A Pydantic dataclass.
         """
+        if is_model_class(cls):
+            raise PydanticUserError(
+                f'Cannot create a Pydantic dataclass from {cls.__name__} as it is already a Pydantic model',
+                code=None,
```

Comment:
Wondering if a new code should be created

Reply:
I think we need a new code